All analyses were performed using the default paramaters for each piece of software, unless otherwise noted.

###Sequence Quality Assessment

Six Illumina (HiSeq2500) gzipped FASTQ sequencing files (100bp, single-end reads; three biological replicates each, from pre- and post-heat shock) were analyzed with FASTQC (v0.11.2). To perform quality trimming, the files were uncompressed with gunzip and the first 15 bases were trimmed with FASTQ/A Trimmer (v0.0.13.2). Upon trimming completion, the FASTQ files were re-compressed with gzip and re-evaluated with FASTQC.

###Sequence Alignment

A Bowtie (v2.1.0) sequence index was created using the <em>Crassostrea gigas</em> genome FASTA file from Ensembl: ```Crassostrea_gigas.GCA_000297895.1.24.dna_sm.genome.fa```. TopHat (v2.0.13.OSX_x86_64), using default parameters was used to align the reads to the Bowtie sequence index.  The two sets of reads (pre- and post-heat shock) were aligned separately. Each TopHat output was directed to seperate directories. The TopHat analysis produced an output file called ```accepted_hits.bam```, which was used in subsequent steps in the RNA-seq workflow.

###Transcriptome Assembly & Quantification

Cufflinks, a component of the Cufflinks computing package (2.2.1.OSX_x86_64), was used to assemble and quantify the reads from the pre- and post-heat shock, separately. The Cufflinks analysis was performed on each ```accepted_hits.bam``` file from the previous TopHat outputs. Additionally, the Cufflinks analyses utilized the Ensembl general transfer format (.gtf) file associated with the Ensembl genome file for sequence annotation information. Each analysis with Cufflinks analysis was directed to new, separate directories and generated a ```transcripts.gtf``` file to be used for combining the two transcriptomes in the next step of the workflow.

###Merge Transcriptomes

Using Cuffmerge, another component of the Cufflink software, the two ```transcripts.gtf``` files from the Cufflinks process were merged together and the output file, ```merged.gtf```, was sent to a new directory. The ```merged.gtf``` file was used in the subsequent step of the analysis pipeline.

###Quantifying Gene and Transcript Expression

Gene and transcript expression was determined using Cuffquant, another component of the Cufflinks software package. The separate alignment files (```accepted_hits.bam```) produced by Cufflinks were used in conjuction with the ```merged.gtf``` file generated by Cuffmerge. Each was run separately and the output file (```abundances.cxb```) was directed to a different output directory and given a unique name.  These two .cxb files were used in the subsequent Cuffdiff step.

###Differential Gene Expression

Differential gene expression was calculated using the Cuffidiff package of the Cufflinks software set. Both .cxb files generated by Cuffmerge and the Ensembl genome .gtf file were used as inputs.


